doc/build-helpers/images/dockertools.section.md

```diff
@@ -36,84 +48,216 @@ buildImage {
+  :::
 
-> **_NOTE:_** If you see errors similar to `Error_Protocol ("certificate has unknown CA",True,UnknownCa)` you may need to add `pkgs.cacert` to `contents`.
+    _Default value:_ `null`.
 
-By default `buildImage` will use a static date of one second past the UNIX Epoch. This allows `buildImage` to produce binary reproducible images. When listing images with `docker images`, the newly created images will be listed like this:
+`fromImageTag` _optional_
 
-```ShellSession
-$ docker images
-REPOSITORY   TAG      IMAGE ID       CREATED        SIZE
-hello        latest   08c791c7846e   48 years ago   25.2MB
-```
+: Used to specify the image within the repository tarball in case it contains multiple images.
+  A value of `null` means that `buildImage` will use the first image available in the repository.
 
-You can break binary reproducibility but have a sorted, meaningful `CREATED` column by setting `created` to `now`.
+  :::{.note}
+  This must be used with `fromImageName`. Using only `fromImageTag` without `fromImageName` will make `buildImage` use the first image available in the repository
+  :::
 
-```nix
-pkgs.dockerTools.buildImage {
-  name = "hello";
-  tag = "latest";
-  created = "now";
-  copyToRoot = pkgs.buildEnv {
-    name = "image-root";
-    paths = [ pkgs.hello ];
-    pathsToLink = [ "/bin" ];
-  };
+    _Default value:_ `null`.
 
-  config.Cmd = [ "/bin/hello" ];
-}
-```
+`copyToRoot` _optional_
 
-Now the Docker CLI will display a reasonable date and sort the images as expected:
+: Files to add to the generated image.
+  This can be either a path or a list of paths.
+  This can be seen as an equivalent of `ADD contents/ /` in a `Dockerfile`.
 
-```ShellSession
-$ docker images
-REPOSITORY   TAG      IMAGE ID       CREATED              SIZE
-hello        latest   de2bf4786de6   About a minute ago   25.2MB
-```
+  If a derivation is used in this attribute, its output path will be used instead.
+  Because of this, it's a common usage to see `pkgs.buildEnv` used in this attribute.
+
+    _Default value:_ `null`.
+
+`keepContentsDirlinks` _optional_
+
+: When adding files to the generated image (as specified by `copyToRoot`), this attribute controls whether to preserve symlinks to directories.
+  If `false`, the symlinks will be transformed into directories.
+  This behaves the same as `rsync -k` when `keepContentsDirlinks` is `false`, and the same as `rsync -K` when `keepContentsDirlinks` is `true`.
+
+    _Default value:_ `false`.
+
+`runAsRoot` _optional_
+
+: A bash script that will run as root inside a VM that contains the existing layers of the base image and the new generated layer (including the files from `copyToRoot`).
+  This can be seen as an equivalent of `RUN ...` in a `Dockerfile`.
+  A value of `null` means that this step in the image generation process will be skipped.
+
+  :::{.caution}
+  Using this attribute requires the `kvm` device to be available.
```

Comment:
```suggestion
  Using this attribute requires the `kvm` device to be available, see [`system-features`](https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-system-features)
```

Reply:
The highlight in this case actually kind of makes sense, I didn't mind keeping it :)

Reply:
Decided to use the link without highlight. After opening the one with highlight again, it felt a bit weird because only 2 words in the whole page were highlighted.